syntax = "proto3";

package roompackage;

enum Type {
    Small = 0;
    Medium = 1;
    Big = 2;
    Large = 3;
}

service RoomService {
    rpc CreateRoom(RoomRequest) returns (RoomResponse);
    //rpc GetRoom(int64) returns (Room);
    //rpc GetRooms() returns (stream Room);
    //rpc UpdateRoom(Room) returns (Room);
};


message RoomResponse {
    string _name                = 1;
    int64 _number               = 2;
    Type _type                  = 3;
    optional int64 _residents   = 4;
}

message RoomRequest {
    string _name                = 1;
    int64 _number               = 2;
}
